src/segmenter/mod.rs

```diff
@@ -214,20 +251,20 @@ impl<'o> Segment<'o> for &'o str {
                 }
                 current_script
             })
-            .map(|s| {
-                let mut detector = s.detect();
+            .map(move |s| {
```

Comment:
it shouldn't be there, because you don't want to move the allow_list.
```suggestion
            .map(|s| {
```
I suggest fixing this one after the other fixes. And if you need help, I can investigate this one.

Reply:
I fixed all else and now investigating, but this is the error I get which suggests adding the move
`
error[E0373]: closure may outlive the current function, but it borrows `allow_list`, which is owned by the current function
   --> src/segmenter/mod.rs:253:18
    |
253 |             .map(|s| {
    |                  ^^^ may outlive borrowed value `allow_list`
254 |                 let mut detector = s.detect(allow_list);
    |                                             ---------- `allow_list` is borrowed here
    |
note: closure is returned here
   --> src/segmenter/mod.rs:262:9
    |
262 |         SegmentedTokenIter::<'o> { inner: Box::new(inner), char_index: 0, byte_index: 0 }
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: to force the closure to take ownership of `allow_list` (and any other referenced variables), use the `move` keyword
    |
253 |             .map(move |s| {
    |                  ++++
`